daemon/cluster/executor/container/controller.go

```diff
@@ -163,8 +164,23 @@ func (r *controller) Start(ctx context.Context) error {
 		return exec.ErrTaskStarted
 	}
 
-	if err := r.adapter.start(ctx); err != nil {
-		return errors.Wrap(err, "starting container failed")
+	for {
+		if err := r.adapter.start(ctx); err != nil {
+			if _, ok := err.(libnetwork.ErrNoSuchNetwork); ok {
+				// Retry network creation again if we
+				// failed because some of the networks
+				// were not found.
+				if err := r.adapter.createNetworks(ctx); err != nil {
+					return err
+				}
+
+				continue
```

Comment:
no need to change this now, but it's typically somewhere i would just use a `goto` :)